File: little-ide/src/main/java/com/quane/little/view/DraggableComponent.java
package com.quane.little.view;

import com.vaadin.ui.Component;
import com.vaadin.ui.DragAndDropWrapper;

/**
 * A draggable UI component.
 * 
 * @author Sean Connolly
 * 
 * @param <T>
 *            the child component type
 */
public class DraggableComponent<T extends Component> extends DragAndDropWrapper {

	private static final long serialVersionUID = 140115L;

	private final T component;

	public DraggableComponent(T component) {
		super(component);
		this.component = component;
		setDragStartMode(DragStartMode.WRAPPER);
		setSizeUndefined();
	}

	public T getComponent() {
		return component;
	}

}

File: little-ide-core/src/main/scala/com/quane/vaadin/scala/DraggableComponent.scala
package com.quane.vaadin.scala

import com.vaadin.ui.DragAndDropWrapper.DragStartMode

import vaadin.scala.Component

/**
 * A draggable UI component.
 *
 * @author Sean Connolly
 *
 * @param <T>
 *         the child component type
 */
class DraggableComponent[C <: Component](c: C) extends DragAndDropWrapper(c) {

  p.setDragStartMode(DragStartMode.WRAPPER);
  p.setSizeUndefined();

}